DAOS-5127 vos: callbacks for pmem_memmove/memcpy

gnailzenh opened the pull request:
A few changes in this patch:
- calls pmem_memmove/memcpy for persistent memory
- no snapshot for newly allocated ilog root
- decrease snapshot size of dbtree and evtree

Signed-off-by: Liang Zhen <liang.zhen@intel.com>

daosbuild1 commented on src/common/mem.c:
```diff
@@ -353,6 +379,70 @@ static umem_ops_t	pmem_ops = {
+{
+	if (!UMOFF_IS_NULL(umoff)) {
+		PMEMoid oid = umem_off2id(umm, umoff);
+		pmemobj_free(&oid);
```
(style)  Missing a blank line after declarations

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2509/1/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-2509/1/execution/node/60/log

daosbuild1 commented:
Test stage Functional completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-2509/1/execution/node/698/log

daosbuild1 commented:
Test stage Functional_Hardware_Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-2509/1/execution/node/764/log

Commit:
DAOS-5127 vos: callbacks for pmem_memmove/memcpy

A few changes in this patch:
- calls pmem_memmove/memcpy for persistent memory
- no snapshot for newly allocated ilog root
- decrease snapshot size of dbtree and evtree

Signed-off-by: Liang Zhen <liang.zhen@intel.com>

daosbuild1 commented on src/common/mem.c:
```diff
@@ -353,6 +379,71 @@ static umem_ops_t	pmem_ops = {
+{
+	if (!UMOFF_IS_NULL(umoff)) {
+		PMEMoid oid = umem_off2id(umm, umoff);
+		pmemobj_free(&oid);
```
(style)  Missing a blank line after declarations

daosbuild1 commented on src/common/mem.c:
```diff
@@ -353,6 +379,71 @@ static umem_ops_t	pmem_ops = {
+		return -DER_INVAL;
+
+	/*
+	 * vmem doesn't support transaction, so we just execute the commit
```
(style)  'doesn'' may be misspelled - perhaps 'doesn't'?

daosbuild1 commented on src/vos/evtree.c:
```diff
@@ -1268,10 +1269,16 @@ evt_node_alloc(struct evt_context *tcx, unsigned int flags,
+	/* NB: have an extra entry because assuming it's an insert */
+	nob = sizeof(struct evt_node) +
+	      sizeof(struct evt_node_entry) *
+	      min(tcx->tc_order, nd->tn_nr + 1);
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/vos/evtree.c:
```diff
@@ -1118,7 +1118,8 @@ evt_desc_log_del(struct evt_context *tcx, daos_epoch_t epoch,
 
 static int
-evt_node_entry_free(struct evt_context *tcx, struct evt_node_entry *ne)
+evt_node_entry_free(struct evt_context *tcx, struct evt_node *nd,
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/vos/evtree.c:
```diff
@@ -1306,7 +1313,7 @@ evt_node_destroy(struct evt_context *tcx, umem_off_t nd_off, int level,
 		if (leaf) {
 			/* NB: This will be replaced with a callback */
-			rc = evt_node_entry_free(tcx, ne);
+			rc = evt_node_entry_free(tcx, nd, ne);
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/vos/evtree.c:
```diff
@@ -2683,7 +2692,7 @@ evt_common_insert(struct evt_context *tcx, struct evt_node *nd,
 				umem_off_t	off = ne->ne_child;
 
-				rc = evt_node_entry_free(tcx, ne);
+				rc = evt_node_entry_free(tcx, nd, ne);
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/vos/evtree.c:
```diff
@@ -2963,8 +2973,8 @@ evt_sdist_split(struct evt_context *tcx, bool leaf, struct evt_node *nd_src,
-	memcpy(entry_dst, entry_src, sizeof(*entry_dst) * (nd_src->tn_nr - nr));
-
+	umem_tx_memcpy(evt_umm(tcx), entry_dst, entry_src,
+		       sizeof(*entry_dst) * (nd_src->tn_nr - nr));
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/vos/evtree.c:
```diff
@@ -1268,10 +1269,16 @@ evt_node_alloc(struct evt_context *tcx, unsigned int flags,
+	nob = sizeof(struct evt_node) +
+	      sizeof(struct evt_node_entry) *
+	      min(tcx->tc_order, nd->tn_nr + 1);
+	return umem_tx_add_ptr(evt_umm(tcx), nd, nob);
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/vos/evtree.c:
```diff
@@ -2660,7 +2669,7 @@ evt_common_insert(struct evt_context *tcx, struct evt_node *nd,
 			 * free the former @desc and re-allocate it properly.
 			 */
-			rc = evt_node_entry_free(tcx, ne);
+			rc = evt_node_entry_free(tcx, nd, ne);
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/vos/evtree.c:
```diff
@@ -2770,8 +2779,8 @@ evt_even_split(struct evt_context *tcx, bool leaf, struct evt_node *nd_src,
-	memcpy(entry_dst, entry_src, sizeof(*entry_dst) * (nd_src->tn_nr - nr));
-
+	umem_tx_memcpy(evt_umm(tcx), entry_dst, entry_src,
+		       sizeof(*entry_dst) * (nd_src->tn_nr - nr));
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/common/btree.c:
```diff
@@ -638,7 +647,12 @@ btr_node_free(struct btr_context *tcx, umem_off_t nd_off)
 btr_node_tx_add(struct btr_context *tcx, umem_off_t nd_off)
 {
-	return umem_tx_add(btr_umm(tcx), nd_off, btr_node_size(tcx));
+	struct btr_node *nd = btr_off2ptr(tcx, nd_off);
```
(style)  'nd' may be misspelled - perhaps 'and'?
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/common/btree.c:
```diff
@@ -1230,7 +1244,7 @@ btr_root_resize(struct btr_context *tcx, struct btr_trace *trace,
 	}
 	trace->tr_node = root->tr_node = nd_off;
-	memcpy(btr_off2ptr(tcx, nd_off), nd, old_size);
+	umem_tx_memcpy(btr_umm(tcx), btr_off2ptr(tcx, nd_off), nd, old_size);
```
(style)  'nd' may be misspelled - perhaps 'and'?
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/common/btree.c:
```diff
@@ -638,7 +647,12 @@ btr_node_free(struct btr_context *tcx, umem_off_t nd_off)
-	return umem_tx_add(btr_umm(tcx), nd_off, btr_node_size(tcx));
+	struct btr_node *nd = btr_off2ptr(tcx, nd_off);
+
+	return umem_tx_add(btr_umm(tcx), nd_off,
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 commented on src/common/btree.c:
```diff
@@ -638,7 +647,12 @@ btr_node_free(struct btr_context *tcx, umem_off_t nd_off)
+	return umem_tx_add(btr_umm(tcx), nd_off,
+			   btr_rec_size(tcx) *
+			   min(tcx->tc_tins.ti_root->tr_node_size,
+			       nd->tn_keyn + 2)); /* assuming insert */
```
(style)  'nd' may be misspelled - perhaps 'and'?

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2509/2/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-2509/2/execution/node/59/log
